onebusaway-android/src/main/java/org/onebusaway/android/ui/ArrivalsListFragment.java

```diff
@@ -268,6 +269,23 @@ public View onCreateView(LayoutInflater inflater,
     }
 
     @Override
+    public void onViewStateRestored(@Nullable Bundle savedInstanceState) {
+        RouteFavoriteDialogFragment dialogFragment = (RouteFavoriteDialogFragment)
+                getFragmentManager().findFragmentByTag(RouteFavoriteDialogFragment.TAG);
+        if (dialogFragment != null) {
+            dialogFragment.setCallback(new RouteFavoriteDialogFragment.Callback() {
+                @Override
+                public void onSelectionComplete(boolean savedFavorite) {
+                    if (savedFavorite) {
+                        refreshLocal();
+                    }
+                }
+            });
+        }
+        super.onViewStateRestored(savedInstanceState);
```

Comment:
Seems like the call to `super.onViewStateRestored(savedInstanceState)` might be better as the first line of this method?  In other words, the super implementation should be called before our code?  Not sure about this.